### ISSUE TITLE
How can I generate sample masks files?

### ISSUE THREAD

**Issue by Reuben-Sun** [3dlg-hcvc/NuiScene#2]
I want to generating some training data by myself. I use the DualOctreeGNN mentioned in the paper and successfully obtained pointcloud.npz and occu.npz, but I am not quite clear on how to obtain the sample_masks.npy files. Could you provide more information?

**Comment by hanhung** [comment]
Hello,

For NuiScene43 if you want to directly get the preprocessed scenes (sample masks, occupancy and point cloud) you can see our [NuiScene43-Datasets](https://github.com/3dlg-hcvc/NuiScene43-Dataset) repo. I also uploaded our tool for converting mesh to occupancy at [taichi_mesh2sdf](https://github.com/hanhung/taichi_mesh2sdf).

I’ll upload the code for point cloud sampling and sample masks soon. The sample mask is generated by first creating an alpha mask from the top-down view of the scene. A convolution with an all-ones kernel is then applied, followed by thresholding to produce the final sampling mask. This ensures that all selected chunks are fully occupied and free of holes or boundary falloffs. For the depth variation map, a custom convolution kernel is used to compute the difference between local window depths and the mean depth. The result is averaged to get the depth variation map which helps avoid sampling overly flat regions.

**Comment by hanhung** [comment]
Hello,

Sorry for the late response. You can find an example of our alpha map generation code here: [https://github.com/3dlg-hcvc/NuiScene43-Dataset?tab=readme-ov-file#notes](https://github.com/3dlg-hcvc/NuiScene43-Dataset?tab=readme-ov-file#notes).
